pkg/controller/volume/attachdetach/reconciler/reconciler.go

```diff
@@ -129,6 +130,19 @@ func (rc *reconciler) syncStates() {
 	}
 }
 
+func (rc *reconciler) isExclusiveVolume(volumeSpec *volume.Spec) bool {
+	if volumeSpec.PersistentVolume == nil {
+		return false
+	}
+	// check if this volume can be attached to multiple PODs/nodes, if yes, return false
+	for _, accessMode := range volumeSpec.PersistentVolume.Spec.AccessModes {
```

Comment:
Ok, now I finally got it. I was not aware of GCE and AWS allowing to attach on multiple nodes with read only. That's why I thought readOnly had nothing to do with attaching. 

I'd suggest to change this function to be called "isMultiAttachForbiden" and start with just a few checks for volume types we definitely know that multi attach does not work. I would start with only Azure disks and then later we can add more here. This way, we make sure that we don't break use cases that worked before and at least fix the problems we have on azure due to double attach. Other volume types which forbid multi attach would still fail when the attach is tried later, but it shouldn't be as bad as on azure (failing faster)